pallets/payment/src/lib.rs

```diff
@@ -333,9 +332,7 @@ pub mod pallet {
 					T::Asset::unreserve(payment.asset, &to, payment.amount);
 					// transfer amount to creator
 					match T::Asset::transfer(payment.asset, &to, &from, payment.amount) {
-						Ok(_) =>
-							*maybe_payment =
-								Some(PaymentDetail { state: PaymentState::Cancelled, ..payment }),
+						Ok(_) => *maybe_payment = None, // delete canceled payment from storage
```

Comment:
One scenario I can think of is like this
- User creates payment and reserves funds
- User transfers balance somehow without keepalive requirement, account gets reaped
- When unreserve and attempt to transfer, the KeepAlive requirements can cause the transfer to fail 

I think its safe to keep the check anyway to handle some weird edge case 😅

Reply:
Still I don't think the best way to handle the failure should be succeeding and leaving the payment in `NeedsReview`, the reviewer can only cancel and release, likely the action to take is to try to cancel again but if we test it further we'll see  we get to a weird state since the unreserve already took place and more errors will pop up. Instead we should probably make sure the account cannot be reaped while a payment is in progress, I haven't used it but I believe we can achieve that with the providers/consumers feature of the system pallet

Reply:
Sure I'll look into that, although looking at similar pallet in polkadot, they just seem to [expect it to not fail](https://github.com/paritytech/polkadot/blob/468a75ed56852afd7d5c161407b3b70d342da64d/runtime/common/src/auctions.rs#L496) like you said. Another idea that I thought of is to transfer the amount to the pallet account, so the pallet really does becomes an escrow, similar to how its done in [treasury pallet](https://github.com/paritytech/substrate/blob/ded44948e2d5a398abcb4e342b0513cb690961bb/frame/treasury/src/lib.rs#L493). Wdyt? 🤔

Reply:
Using the pallet account would be okish with the named reserves but just adds extra complexities and losing the feature of users being able to proof they have some "money on the way". Let's do the expectation and create a separate issue to protect ourselves from that edge-case by protecting the account from being deleted while a payment is in progress(i.e. we increase the ref count in system pallet when payment is created and remove when payment is released/canceled)

Reply:
For reference https://docs.substrate.io/v3/advanced/account-info/#account-reference-counters 
Now I'm more convinced that's the way to go, in a way people are temporarily paying the pallet to store some data about themselves(payments) and with the reference counters mechanism we tap into the account managing system forcing users to resolve all their pending payments before they can reap their account and take the existential deposit back

Reply:
#175